src/HotChocolate/Core/src/Validation/Rules/FieldVisitor.cs

```diff
@@ -370,7 +370,7 @@ private static bool AreArgumentsIdentical(FieldNode fieldA, FieldNode fieldB)
 
                 if (BySyntax.Equals(argumentA.Name, argumentB.Name))
                 {
-                    if (BySyntax.Equals(argumentA.Value, argumentB.Value))
+                    if (AreValuesIdentical(argumentA.Value, argumentB.Value))
```

Comment:
@michaelstaib I wasn't sure of whether to implement the comparision in the SyntaxComparer or outside of it. I assumed outside of it, since the SyntaxComparer compares 1:1 matches, right?
I was just thinking that maybe it would be smarter to implement this as part of the SyntaxComparer? Otherwise things like the Fusion InlineFragmentRewriter also have to reimplement this!?

Reply:
it should go in the syntax comparer as the BySyntax comparison means that the code is the same. The value in this case is the same to the system.